op-batcher/batcher/channel.go

```diff
@@ -52,7 +52,7 @@ func (c *channel) TxFailed(id string) {
 		// Note: when the batcher is changed to send multiple frames per tx,
 		// this needs to be changed to iterate over all frames of the tx data
 		// and re-queue them.
-		c.channelBuilder.PushFrames(data.Frames()...)
+		c.channelBuilder.RewindFrameCursor(data.Frames()[0])
```

Comment:
This is worth a comment to explain why we rewind to the first frame of the transaction which failed.

Reply:
[753ad8e](https://github.com/ethereum-optimism/optimism/pull/12390/commits/753ad8ee795d5623ebd97824a30b081849b14add)